internal/controller/managedcluster_controller.go

```diff
@@ -337,17 +331,32 @@ func (r *ManagedClusterReconciler) Update(ctx context.Context, managedCluster *h
 			})
 		}
 
-		requeue, err := r.setStatusFromClusterStatus(ctx, managedCluster)
-		if err != nil {
-			if requeue {
-				return ctrl.Result{RequeueAfter: DefaultRequeueInterval}, err
-			}
-
-			return ctrl.Result{}, err
+		childObjects := []schema.GroupVersionResource{
+			{
+				Group:    "cluster.x-k8s.io",
+				Version:  "v1beta1",
+				Resource: "clusters",
+			},
+			{
+				Group:    "cluster.x-k8s.io",
+				Version:  "v1beta1",
+				Resource: "machinedeployments",
+			},
 		}
 
-		if requeue {
-			return ctrl.Result{RequeueAfter: DefaultRequeueInterval}, nil
+		for _, gvr := range childObjects {
```

Comment:
I think we should try to collect conditions from all requested objects before we return with an error